charmtools/build/tactics.py

```diff
@@ -26,12 +26,27 @@ class Tactic(object):
     """
     kind = "static"  # used in signatures
 
-    def __init__(self, entity, current, target, config):
+    @classmethod
+    def get(cls, entity, target, layer, next_config, existing_tactic):
+        """
+        Factory method to get an instance of the correct Tactic to handle the
+        given entity.
+        """
+        for candidate in next_config.tactics + DEFAULT_TACTICS:
+            if candidate.trigger(entity, target, layer, next_config):
+                tactic = candidate(entity, target, layer, next_config)
```

Comment:
candidate is a class not a function, so it is guaranteed to return an instance (this is basically like `IgnoreTactic(entity, target, layer, next_config)`).